### PR TITLE
Impro-1121: Remove use of source_realizations

### PR DESCRIPTION
**Pull request by markysparks**
Addresses #GitHubissuenum

Remove Improver use of `source_realizations` attribute as part of IMPRO-1121 work. Note that I have left a `source_realizations` check within` improver/lib/improver/nbhood/nbhood.py` and associated unit test. Whilst this could be removed, I thought maybe it wouldn't do any harm to leave in, but open to comments/suggestions.

https://github.com/markysparks/improver/blob/194631a82cb38942eb0332c2852d98c9d70ccefa/lib/improver/nbhood/nbhood.py#L164

A weighted-blending CLI test that checked that` source_realizations `was present, has been removed and the subsequent tests renumbered. 

Testing:
 - [x] Ran tests and they passed OK
 - [x] Removed unused tests and updated others

### CODE REVIEW

File: tests/improver-weighted-blending/13-source_realizations.bats [hunk 1]
```diff
@@ -1,50 +0,0 @@
-#!/usr/bin/env bats
```
Comment (fionaRust) [inline tests/improver-weighted-blending/13-source_realizations.bats]: I don't think we should remove this test, just update the kgo for it

Comment (markysparks) [reply tests/improver-weighted-blending/13-source_realizations.bats]: Fair enough. I have renamed the bats test as suggested and updated the KGO. Note that the commit below (5c8de16) is referring to removal of a superfluous check in nbhood.py which the previous reviewer requested removed, not this test.

**Comment by bayliffe** [comment]
I have re-run all the tests and can confirm these all pass.

**Comment by fionaRust** [comment]
This is ready but waiting for all PRs then merging all together.

**Comment by bayliffe** [comment]
I have updated the acceptance test KGO that contained source_realizations. I believe this is now ready to merge.

**Pull request merged**